### PR TITLE
feat(lemon-ui): Add `LemonDropdown` and `LemonMenu`

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

We have lots of menus, and they are all implemented like this:

```TypeScript
<LemonButtonWithDropdown
    {/* some props */}
    dropdown={{
        overlay: (
            <>
                <LemonButton status="stealth" to={...} fullWidth>
                    View
                </LemonButton>
                <LemonButton status="stealth" to={...} fullWidth>
                    Edit
                </LemonButton>
                {/* et cetera et cetera */}
            </>
        )
    }}
/>
```

This is all too bespoke! That's a standard menu pattern which should be handled via a common abstraction.
Previously this was mostly just extra mental load, but with PostHog 3000 we need more menus generated in logics as opposed to concrete React components (e.g. right-click menus for sidebar items, or page-level action menus).

## Changes

We already had a lot of menu-like handling in `LemonSelect` (in particular keyboard navigation was very concrete), so I borrowed all that for the new `LemonMenu` component. On the way, I extracted some "show/hide dropdown" logic from `LemonButtonWithDropdown` into `LemonDropdown`. Then I rebuilt `LemonSelect` on top of `LemonMenu` to verify the new abstraction.

This is the new hierarchy:
1. `Popover` is the raw building block for any popover in the UI – it handles positioning, but does not control whether its visible or not
2. `LemonDropdown` builds on top of `Popover` to abstract away handling of open/closed state
3. `LemonMenu` builds on top of `LemonDropdown` to abstract away overlay contents for the common menu case (and behavior tied to this use case, e.g. keyboard navigation)
4. `LemonSelect` builds on top of `LemonMenu` to abstract away `value` and `onChange` semantics
 
EDIT: Extracted the `LemonSelect` refactor out into #15116.

## How did you test this code?

Relying on visual regression tests here.

### CODE REVIEW

**Comment by benjackwhite** [comment]
Not here to review - just to say - YYYESSSSS!!!

**Comment by Twixes** [comment]
But while you're here, feel free to review @benjackwhite ;)

File: frontend/src/lib/lemon-ui/Popover/Popover.tsx [hunk 1]
```diff
@@ -51,18 +45,16 @@ export interface PopoverProps {
      * Works also with strings, matching classnames or ids, for antd legacy components that don't support refs
      * **/
     additionalRefs?: (React.MutableRefObject<HTMLDivElement | null> | string)[]
-    style?: React.CSSProperties
-    /** Whether the parent popover should be closed as well on click. Useful for menus  */
```
Comment (thmsobrmlr) [inline frontend/src/lib/lemon-ui/Popover/Popover.tsx:55]: Did you remove this JSDoc comment on purpose?

**Review by thmsobrmlr — APPROVED** [review]
Yay, for this! Can't believe we've reinvented the wheel over and over until now.

A few (minor) comments:
- It could get annoying (and surprising!) that any button nested within a popover gets the dropdown arrow. Maybe have a prop on the popover level that then gets propagate through context?
- It's a bit strange that the storybook shows usage of the `LemonMenuPopover` and not of `LemonMenu` itself,  but I get that you want to display (and snapshot) the actual menu. Could just be confusing for new people looking how implement menu and use Storybook for this use case. Maybe add some markdown help text?
- The keyboard navigation state looks a bit strange to me with the border being different size between x- and y-axis. 
  Screenshot 2023-04-19 at 18 09 10

File: frontend/src/lib/lemon-ui/Popover/Popover.tsx (hunk 1 above)
Comment (Twixes) [reply frontend/src/lib/lemon-ui/Popover/Popover.tsx:55]: Oops, this got mangled a bit, fixed!

**Comment by Twixes** [comment]
Thanks for the feedback. My thoughts:
1. I agree the default dropdown icon might be annoying. The idea of it it seems like a useful default in _most cases_. We had it in `LemonButtonWithDropdown` before (which we'll move away from), so that's why it's not a bit broader – I'm gonna leave it in for now, and see if there are any more cases where we don't want it (besides those "Add filter" buttons, which I gave some overrides).
2. Added a description to the story to explain `LemonMenuOverlay` usage.
3. Weird, I can't reproduce the focus issue. 👀 I'll revisit that after #15116 is in though, as overall keyboard navigation might need a bit of polish down the line.

**Pull request merged**